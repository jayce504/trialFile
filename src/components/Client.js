import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';

class Client extends Component {
    constructor() {
        super()

        this.state = { 
            name: '', 
            address: '',
            phoneNumber: '',
            allegedOffenses: '',
            offenseClass: '',
            disposition: '',
            sentence:'',
            sentencingJudge:''
        }
    }

    render() {
        return(
            <div className='client'>
                <Form>
                    <FormGroup>
                        <ControlLabel>Name:</ControlLabel>
                        <FormControl 
                            className='input-person' 
                            onChange={event => this.setState({ name: event.target.value })}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Address:</ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {address: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Phone Number:</ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {phoneNumber: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Alleged Offenses:
                        </ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {allegedOffenses: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Offense Class:
                        </ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {offenseClass: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Disposition
                        </ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {disposition: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Sentence: 
                        </ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {sentence: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <ControlLabel>Sentencing Judge: 
                        </ControlLabel>
                        <FormControl
                        className='input-person'
                        onChange={event => this.setState( {sentencingJudge: event.target.value})}
                        />
                    </FormGroup>
                </Form>
                <Button
                    className='btn-remove'
                    onClick={() => this.props.removeClient(this.props.client.id)}
                    >
                    Remove Client
                </Button>
            </div>
        )
    }
}

export default Client;