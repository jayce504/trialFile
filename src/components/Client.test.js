import React from 'react';
import { shallow } from 'enzyme';
import Client from "./Client";

describe('Client', () => {
    const mockRemove = jest.fn()
    const id = 1
    const props = { client: { id }, removeClient: mockRemove }
    const client = shallow(<Client {...props} />);

    it('renders properly', () => {
        expect(client).toMatchSnapshot();
    });

    it('initializes a client and related info in the `state`', () => {
        expect(client.state()).toEqual({ name: '', address:'', phoneNumber:'', allegedOffenses:'' })
    });

    describe('when typing into client input', () => {
        const name = ''
        const address = 'Address 1'
        const phoneNumber = '111-111-1111'
        const allegedOffenses = 'some'
        const client = shallow(<Client {...props} />);

        beforeEach(() => {
            client.find('.input-person').simulate('change', { target: {value: name } })
        })

        beforeEach(() => {
            client.find('.input-person').simulate('change', { target: {value: address } })
        })

        beforeEach(() => {
            client.find('.input-person').simulate('change', { target: {value: phoneNumber } })
        })

        beforeEach(() => {
            client.find('.input-person').simulate('change', { target: {value: allegedOffenses } })
        })

        it('updates the client name in `state`', () => {
            expect(client.state().name).toEqual(name)
        })
    });

    describe('when clicking the `Remove Client` button', () => {
        client.find('.btn-remove').simulate('click')
    });
        
    it('calls the removeClient callback', () => {
            expect(mockRemove).toHaveBeenCalledWith(id)
        });
});   
