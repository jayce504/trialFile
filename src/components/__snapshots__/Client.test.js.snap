// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Client
    client={
      Object {
        "id": 1,
      }
    }
    removeClient={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Name:
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Address:
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Phone Number:
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Alleged Offenses:
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>
        </Form>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn-remove"
          disabled={false}
          onClick={[Function]}
        >
          Remove Client
        </Button>,
      ],
      "className": "client",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Name:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Address:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Phone Number:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Alleged Offenses:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Name:
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Name:",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Name:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Address:
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Address:",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Address:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Phone Number:
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Phone Number:",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Phone Number:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Alleged Offenses:
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Alleged Offenses:",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Alleged Offenses:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Remove Client",
          "className": "btn-remove",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Client",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Name:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Address:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Phone Number:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Alleged Offenses:
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />
            </FormGroup>
          </Form>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-remove"
            disabled={false}
            onClick={[Function]}
          >
            Remove Client
          </Button>,
        ],
        "className": "client",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Name:
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />
              </FormGroup>,
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Address:
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />
              </FormGroup>,
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Phone Number:
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />
              </FormGroup>,
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Alleged Offenses:
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                />
              </FormGroup>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Name:
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Name:",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Name:",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Address:
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Address:",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Address:",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Phone Number:
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Phone Number:",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Phone Number:",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Alleged Offenses:
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Alleged Offenses:",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Alleged Offenses:",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Remove Client",
            "className": "btn-remove",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Client",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
